discovery_engine_core/web-api/src/handlers.rs

```diff
@@ -21,15 +22,18 @@ use crate::{
     models::{InteractionRequestBody, UserId},
 };
 
+#[instrument(skip(db))]
 pub(crate) async fn handle_ranked_documents(
     user_id: UserId,
     db: Db,
 ) -> Result<impl warp::Reply, Rejection> {
-    let user_interests = db
-        .user_state
-        .fetch(&user_id)
-        .await
-        .map_err(handle_user_state_op_error)?;
+    let user_interests = db.user_state.fetch(&user_id).await.map_err(|err| {
+        error!(
+            "Error fetching interests for user '{}': {:#?}",
+            user_id, err
+        );
+        handle_user_state_op_error(err)
+    })?;
 
     let mut documents = db.documents.clone();
 
```

Comment:
are we interested also how much time ranking takes? @acrrd

Reply:
In the whole request. We can do it here but maybe we can do it in a more comprehensive way from the load balancer.
we can avoid it for a moment and understand later what to do.